File: prog3/trab_java/utilitarios/capes/producoes/Anal.java
package utilitarios.capes.producoes;


public class Anal extends Producao implements Comparable<Anal>{
    private String evento;

    public Anal(String titulo, String natureza, String idioma, String cidade, int qtdPags, String evento){
        super(titulo, natureza, idioma, cidade, qtdPags);
        this.evento = evento;
        this.tipo = 1;
    }  

    public static void imprimeCabecalho(){
        System.out.println("Natureza;Titulo;Idioma;Evento;Cidade;Paginas");
    }
    
    @Override
    public String toString(){
        return (this.natureza + ";" + this.titulo + ";" + this.idioma + ";" + this.evento + ";" + this.cidade + ";" + this.qtdPags_string);
    }
    
    @Override
    public int compareTo(Anal o) {
        return this.natureza.compareTo(o.natureza) != 0 ?
        this.natureza.compareTo(o.natureza) : this.titulo.compareTo(o.titulo) != 0 ?
        this.titulo.compareTo(o.titulo) : this.idioma.compareTo(o.idioma) != 0 ?
        this.idioma.compareTo(o.idioma) : this.evento.compareTo(o.evento) != 0 ?
        this.evento.compareTo(o.evento) : this.cidade.compareTo(o.cidade) != 0 ?
        this.cidade.compareTo(o.cidade) : (this.qtdPags - o.qtdPags);
    }
}

File: prog3/trab_cpp/Anal.cpp
#include "Anal.hpp"
#include <exception>

Anal::Anal(string titulo, string natureza, string idioma, string cidade, int qtd_pags, string evento): Producao(titulo, natureza, idioma, cidade, qtd_pags, 8){
    this->evento = evento;
}

Anal::~Anal(){
}

/* Metodo que retorna a string que contem as informacoes de uma producao em formato csv. No caso, reorna uma string no seguinte formato:
 "<Natuzrea>;<Titulo>;<Idioma>;<Evento>;<Cidade>;<Paginas>"*/
string Anal::csv(){
    stringstream stream;
    stream << natureza << ";" << titulo << ";" << idioma << ";" << evento << ";" << cidade << ";" << _int_string(qtd_pags);
    return stream.str();
}
/* Metodo que retorna a string cabecalho. */
string Anal::cabecalho(){
    return "Natureza;Titulo;Idioma;Evento;Cidade;Paginas";
}

/* Operador de comparacao de uma producao com outra de mesmo subtipo. */
bool Anal::operator < (Producao& o){
    Anal& prod = (Anal&)o;
    int cmp = natureza.compare(prod.natureza);
    if(cmp<0) return true; else if(cmp>0) return false;
    cmp = titulo.compare(prod.titulo);
    if(cmp<0) return true; else if(cmp>0) return false;
    cmp = idioma.compare(prod.idioma);
    if(cmp<0) return true; else if(cmp>0) return false;
    cmp = evento.compare(prod.evento);
    if(cmp<0) return true; else if(cmp>0) return false;
    cmp = cidade.compare(prod.cidade);
    if(cmp<0) return true; else if(cmp>0) return false;
    cmp = qtd_pags - prod.qtd_pags;
    if(cmp<0) return true; else return false;
}
